src/main/java/ui/listpanel/ListPanel.java

```diff
@@ -212,10 +212,20 @@ private void setupKeyboardShortcuts() {
             if (NEW_COMMENT.match(event)) {
                 if (KeyPress.isValidKeyCombination(GOTO_MODIFIER.getCode(), event.getCode())) {
                     ui.getBrowserComponent().switchToTab(DISCUSSION_TAB);
-                } else if (ui.getBrowserComponent().isCurrentUrlIssue()) {
-                    ui.getBrowserComponent().switchToTab(DISCUSSION_TAB);
-                    ui.getBrowserComponent().jumpToComment();
+                } else {
+                    if (!ui.getBrowserComponent().isCurrentUrlPrDiscussion()) {
+                        ui.getBrowserComponent().switchToTab(DISCUSSION_TAB);
+                    }
+
+                    Thread jumpToCommentOncePageLoaded = new Thread(() -> {
+                        while (!ui.getBrowserComponent().isCurrentUrlPrDiscussion()) {
+                        // wait
+                        }
+                        ui.getBrowserComponent().jumpToComment();
+                    });
+                    jumpToCommentOncePageLoaded.start();
```

Comment:
Met him just now. Will use Thread.sleep.
On Mar 17, 2016 3:41 PM, "Nguyen Quoc Dat" <notifications@github.com> wrote:

> In src/main/java/ui/listpanel/ListPanel.java
> <https://github.com/HubTurbo/HubTurbo/pull/1367#discussion_r56465500>:
>
> > @@ -212,10 +212,20 @@ private void setupKeyboardShortcuts() {
> >              if (NEW_COMMENT.match(event)) {
> >                  if (KeyPress.isValidKeyCombination(GOTO_MODIFIER.getCode(), event.getCode())) {
> >                      ui.getBrowserComponent().switchToTab(DISCUSSION_TAB);
> > -                } else if (ui.getBrowserComponent().isCurrentUrlIssue()) {
> > -                    ui.getBrowserComponent().switchToTab(DISCUSSION_TAB);
> > -                    ui.getBrowserComponent().jumpToComment();
> > +                } else {
> > +                    if (!ui.getBrowserComponent().isCurrentUrlPrDiscussion()) {
> > +                        ui.getBrowserComponent().switchToTab(DISCUSSION_TAB);
> > +                    }
> > +
> > +                    Thread jumpToCommentOncePageLoaded = new Thread(() -> {
> > +                        while (!ui.getBrowserComponent().isCurrentUrlPrDiscussion()) {
> > +                        // wait
> > +                        }
> > +                        ui.getBrowserComponent().jumpToComment();
> > +                    });
> > +                    jumpToCommentOncePageLoaded.start();
>
> Thread.sleep would be better, I think. @dariusf
> <https://github.com/dariusf> any comments?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/HubTurbo/HubTurbo/pull/1367/files/d731abf1d17523e5e0395c456dc58c2792741f74#r56465500>
>